clean.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b         Delete bin directory"
+    echo "  -p         Delete packages directory"
+    echo "  -t         Delete Tools directory"
+    echo "  -w         Remove all untracked files in the working tree, including the bin, packages, and Tools directories"
+    echo "  -c         Delete the user-local NuGet package caches, located in ~/.local/share/dnu and ~/.local/share/NuGet"
+    echo "  -a, --all  Clean all of the above"
+    exit 1
+}
+
+invoke_buildproj()
+{
+    PackageRoot=$WorkingTreeRoot/packages
+    MSBuildPackageId="Microsoft.Build.Mono.Debug"
+    MSBuildPackageVersion="14.1.0.0-prerelease"
+    MSBuildPath=$PackageRoot/$MSBuildPackageId.$MSBuildPackageVersion/lib/MSBuild.exe
+
+    MONO29679=1 mono $MSBuildPath $WorkingTreeRoot/build.proj /nologo /verbosity:minimal /t:$1
+}
+
+WorkingTreeRoot="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Parse arguments
+
+if [ $# == 0 ]
+then
+    echo "Please specify at least one option."
+    echo "Use 'clean -h' for help."
+    exit 1
+fi
+
+while [[ $# > 0 ]]
+do
+    opt="$1"
+    case $opt in
+        -h|--help)
+        usage
+        ;;
+        -b)
+        CleanBinDir=true
+        ;;
+        -p)
+        CleanPackagesDir=true
+        ;;
+        -t)
+        CleanToolsDir=true
+        ;;
+        -w)
+        CleanWorkingTree=true
+        ;;
+        -c)
+        CleanPackagesCache=true
+        ;;
+        -a|--all)
+        CleanWorkingTree=true
+        CleanPackagesCache=true
+        ;;
+        *)
+        echo "Unrecognized argument '$opt'"
+        echo "Use 'clean -h' for help."
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+if [ "$CleanPackagesCache" == true ]
+then
+    echo "Removing the user-local NuGet package caches"
+    invoke_buildproj CleanPackagesCache
+fi
+
+if [ "$CleanBinDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing bin directory"
+    invoke_buildproj Clean
+fi
+
+if [ "$CleanPackagesDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing packages directory"
+    invoke_buildproj CleanPackages
+fi
+
+if [ "$CleanToolsDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing Tools directory"
+    # This directory cannot be removed in a build target because MSBuild is in the Tools directory
+    rm -rf $WorkingTreeRoot/Tools
+fi
+
+if [ "$CleanWorkingTree" == true ]
+then
+    echo "Removing all untracked files in the working tree"
+    git clean -xdfq $WorkingTreeRoot
+fi
+
+echo "Clean complete!"
```

Comment:
Good point. I've changed it so that it checks the exit status of each command to see if there was an error.